src/DBus.vala

```diff
@@ -256,5 +296,125 @@ namespace Gala {
+         * 
+         * Calling this method will create a static temporary snapshot from a target window
+         * and will transition from it to the real window surface. This is useful if
+         * e.g you want to change the Gtk stylesheet used by your application.
+         * It can be used to create a smooth transition effect when switching to the new
+         * stylesheet. If that's your intention, call this function just before actually changing
+         * the stylesheet for the best effect.
+         * 
+         * The function will create a transition animation only when animations are enabled.
+         * If they are not, the animation will not be visible but the snapshot will be still created and added.
+         * 
+         * The transition will be applied to all windows that belong to your process ID. 
+         * 
+         * Note that you will not be able to affect any windows that do
+         * not belong to your PID, this is reserved for system components.
+         */
+        public async void transition_from_snapshot (BusName sender) throws IOError {
+            if (bus_proxy == null) {
+                throw new IOError.FAILED ("Could not connect to org.freedesktop.DBus");
+            }
+
+            uint32 pid = bus_proxy.get_connection_unix_process_id (sender);
+            bool animate = wm.enable_animations;
+
+            foreach (unowned Meta.WindowActor actor in wm.get_display ().get_window_actors ()) {
+                unowned Meta.Window window = actor.get_meta_window ();
+                if (window.get_pid () == pid) {
+                    yield transition_window (actor.get_meta_window (), animate);
+                }
+            }
+        }
+
+        /**
+         * Creates a window snapshot and transitions from it to the real state
+         * for all windows on the visible workspace. See transition_from_snapshot.
+         * 
+         * This function is reserved for system components and must not be
+         * accessed by applications.
+         */
+        public async void global_transition_from_snapshot () throws IOError {
+            bool animate = wm.enable_animations;
+
+            unowned Meta.Workspace workspace = wm.get_display ().get_workspace_manager ().get_active_workspace ();
+            var windows = workspace.list_windows ();
+            foreach (unowned Meta.Window window in windows) {
+                if (!window.minimized && window.get_window_type () != Meta.WindowType.DESKTOP) {
+                    yield transition_window (window, animate);
+                }
+            }
+        }
+
+        static async void transition_window (Meta.Window window, bool animate) {
+            /**
+             * Because Gtk takes time to apply the styling to all elements,
+             * we will simply define an arbitrary wait time for this to happen
+             * to smoothen out the transition.
+             */
+            const uint STYLE_CHANGE_COMPLETION_DURATION = 150;
```

Comment:
I've been playing with this, and 300 seems to be long enough to mask WebKit transitioning as well which is a really nice effect.

```suggestion
            const uint STYLE_CHANGE_COMPLETION_DURATION = 300;
```